qst/src/main/java/io/deephaven/qst/table/HeadTable.java

```diff
@@ -0,0 +1,34 @@
+package io.deephaven.qst.table;
+
+import io.deephaven.qst.NodeStyle;
+import org.immutables.value.Value.Check;
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Parameter;
+
+@Immutable
+@NodeStyle
+public abstract class HeadTable extends TableBase implements SingleParentTable {
+
+    public static HeadTable of(Table parent, long size) {
+        return ImmutableHeadTable.of(parent, size);
+    }
+
+    @Parameter
+    public abstract Table parent();
+
+    @Parameter
+    public abstract long size();
+
+    @Override
+    public final <V extends Visitor> V walk(V visitor) {
+        visitor.visit(this);
+        return visitor;
+    }
+
+    @Check
+    final void checkSize() {
+        if (size() <= 0) {
```

Comment:
Engine allows 0. I'm not sure whether we should, or shouldn't. Tail, too.

Reply:
I'm inclined to save the user from themselves in this case, unless we can think of why this would be useful.

Reply:
I'll ask @cpwright if he knows of a reason to allow 0. I'd prefer QST match engine where reasonable, but you may have a point here.